src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -257,6 +257,12 @@ private function validateAndDenormalize($currentClass, $attribute, $data, $forma
                 if (null !== $collectionKeyType = $type->getCollectionKeyType()) {
                     $context['key_type'] = $collectionKeyType;
                 }
+            } elseif ($type->isCollection() && null !== $type->getCollectionValueType()) {
```

Comment:
I'd suggest merging these conditions with the next if:
`if ('xml' === $format && !\is_array($data) && $type->isCollection() && null !== $type->getCollectionValueType()) {`

Reply:
Thank you for your feedback and all your work! Done...